go/types/opcache.go

```diff
@@ -122,6 +173,45 @@ func (i *opCacheIterator) Op() sequenceItem {
 	return entry
 }
 
-func (i *opCacheIterator) Release() {
+// Insert can be called from any goroutine
+func (opc *LdbOpCache) SetInsert(val Value) {
+	valArray := [initialBufferSize]byte{}
+
+	switch val.Type().Kind() {
+	default:
+		// Set entries are fairly simple to store, simply store the element's hash (or the element itself if bool, number, or string) in the ldb key and the value in ldb value.
+
+		// Note that, if val is a prolly trees, any in-memory child chunks will be written to vrw at this time.
+		ldbKeyArray := [uint32Size + 1 + hash.ByteLen]byte{}
+		binary.LittleEndian.PutUint32(ldbKeyArray[:], opc.colId)
+		ldbKeyArray[uint32Size] = byte(val.Type().Kind())
+		ldbKeySlice := ldbKeyArray[0 : uint32Size+1]
+		ldbKeySlice = append(ldbKeySlice, val.Hash().DigestSlice()...)
+		encodedVal := encToSlice(val, valArray[:], opc.vrw)
+
+		// TODO: Will manually batching these help?
+		err := opc.store.ldb.Put(ldbKeySlice, encodedVal, nil)
+		d.Chk.NoError(err)
+
+	case BoolKind, NumberKind, StringKind:
+		// In this case, the ldbKey is just the colId + encodedVal, and the ldbValue is just the encodedVal
+		ldbKeyArray := [initialBufferSize]byte{}
+		binary.LittleEndian.PutUint32(ldbKeyArray[:], opc.colId)
+		ldbKeySlice := ldbKeyArray[0:uint32Size]
+		encodedVal := encToSlice(val, valArray[:], opc.vrw)
+		ldbKeySlice = append(ldbKeySlice, encodedVal...)
+		// TODO: Will manually batching these help?
+		err := opc.store.ldb.Put(ldbKeySlice, encodedVal, nil)
```

Comment:
`encodedVal` winds up in both the key and value in the level db, then? I figure you could just put it in the key, and then extract it from there when `SetOp` is called, and save storage. Probably only really meaningful savings in the case of long strings, but still.